sdv/sampling/independent_sampler.py

```diff
@@ -0,0 +1,166 @@
+"""Independent Samplers."""
+import logging
+from copy import deepcopy
+
+import pandas as pd
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseIndependentSampler():
+    """Independent sampler mixin.
+
+    Args:
+        metadata (sdv.metadata.multi_table.MultiTableMetadata):
+            Multi-table metadata representing the data tables that this sampler will be used for.
+        table_synthesizers (dict):
+            Dictionary mapping each table to a synthesizer. Should be instantiated and passed to
+            sampler by the multi-table synthesizer using this sampler.
+        table_sizes (dict):
+            Dictionary mapping each table to its size. Should be instantiated and passed to the
+            sampler by the multi-table synthesizer using this sampler.
+    """
+
+    def __init__(self, metadata, table_synthesizers={}, table_sizes={}):
+        self.metadata = metadata
+        self._table_synthesizers = table_synthesizers
+        self._table_sizes = table_sizes
+
+    def _add_foreign_key_column(self, child_table, parent_table, child_name, parent_name):
+        """Add the foreign key that connects the child table to the parent table.
+
+        Args:
+            child_table (pd.DataFrame):
+                The table containing data sampled for the child.
+            parent_table (pd.DataFrame):
+                The table containing data sampled for the parent.
+            parent_name (str):
+                The name of the parent table.
+            child_name (str):
+                The name of the child table.
+        """
+        raise NotImplementedError()
+
+    def _get_all_foreign_keys(self, table_name):
+        foreign_keys = []
+        for relation in self.metadata.relationships:
+            if table_name == relation['child_table_name']:
+                foreign_keys.append(deepcopy(relation['child_foreign_key']))
+
+        return foreign_keys
```

Comment:
Do we need this to be here or in the `Base` ? Is it used for fitting or only for sampling ?

Reply:
I think ideally this and `get_foreign_keys` (which returns the foreign keys from one parent/child pair) should be _metadata_ methods. Both of them are used for both sampling and fitting. Not sure if that change should happen in these PRs though, it might make sense to put it off until we actually do the integration.

Reply:
I think we should move to the metadata